main.tf

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ */
+
+resource "aws_sqs_queue" "shuntingyard_sqs_queue" {
+   name = "${local.instance_alias}-sqs-queue"
+}
+
+resource "aws_sns_topic_subscription" "sqs_hive_metastore_sns_subscription" {
+  topic_arn = "${var.metastore_events_sns_topic}"
+  protocol  = "sqs"
+  endpoint  = "${aws_sqs_queue.shuntingyard_sqs_queue.arn}"
+}
+
+resource "aws_ecs_cluster" "shuntingyard" {
+  name = "${local.instance_alias}"
+}
+
+resource "aws_iam_role" "shuntingyard_task_exec" {
+  name = "${local.instance_alias}-ecs-task-exec-${var.aws_region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "task_exec_managed" {
+  role       = "${aws_iam_role.shuntingyard_task_exec.id}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "shuntingyard_task" {
+  name = "${local.instance_alias}-ecs-task-${var.aws_region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "s3_for_shuntingyard" {
+  name  = "s3"
+  role  = "${aws_iam_role.shuntingyard_task.id}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:DeleteObject",
+                "s3:DeleteObjectVersion",
+                "s3:Get*",
+                "s3:List*",
+                "s3:PutBucketLogging",
+                "s3:PutBucketNotification",
+                "s3:PutBucketVersioning",
+                "s3:PutObject",
+                "s3:PutObjectAcl",
+                "s3:PutObjectTagging",
+                "s3:PutObjectVersionAcl",
+                "s3:PutObjectVersionTagging"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "sqs_for_shuntingyard" {
+  name  = "sqs"
+  role  = "${aws_iam_role.shuntingyard_task.id}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Effect": "Allow",
+        "Action": [
+          "sqs:SendMessage",
+          "sqs:ReceiveMessage",
+          "sqs:DeleteMessage"
+        ],
+        "Resource": "*"
```

Comment:
probably need to just allow sqs created, instead of all

Reply:
Good point. I have done this.